influent-transport/src/main/java/influent/internal/util/Inet4Network.java

```diff
@@ -0,0 +1,41 @@
+package influent.internal.util;
+
+import java.net.InetAddress;
+import java.util.BitSet;
+
+public class Inet4Network implements InetNetwork {
+  private BitSet netmask;
+  private BitSet network;
+
+  public Inet4Network(InetAddress base, int netmask) {
+    BitSet subnetmask = new BitSet(32);
+    subnetmask.set(32 - netmask, 32);
+    // base.getAddress() returns like [192, 168, 1, 10]
+    // but BitSet.valueOf() expects [10, 1, 168, 192]
+    byte[] reversedBase = InetNetwork.reverseAddress(base);
+    BitSet network = BitSet.valueOf(reversedBase);
+    network.and(subnetmask);
+
+    this.netmask = subnetmask;
+    this.network = network;
+  }
+
+  @Override
+  public BitSet getNetwork() {
+    return network;
+  }
+
+  @Override
+  public BitSet getNetmask() {
+    return netmask;
+  }
+
+  @Override
+  public boolean contains(InetAddress address) {
+    try {
+      return network.equals(InetNetwork.maskIP(address, netmask));
```

Comment:
Excuse me, I have overlooked that `BitSet#equals` ignores sizes.
https://docs.oracle.com/javase/jp/8/docs/api/java/util/BitSet.html#equals-java.lang.Object-

Since masked IPv6 address may matches IPv4 network address, check the version by `address instanceof Inet4Address`.

Reply:
Should `Inet6Network#contains` consider the version same as `Inet4Network#contains`?